src/strategies/erc721-with-metadata-attribute/index.ts

```diff
@@ -0,0 +1,100 @@
+import { Multicaller } from '../../utils';
+import { BigNumber } from '@ethersproject/bignumber';
+import fetch from 'cross-fetch';
+
+export const author = 'allmysmarts';
+export const version = '0.1.0';
+
+const abi = [
+  'function balanceOf(address account) external view returns (uint256)',
+  'function tokenOfOwnerByIndex(address owner, uint256 index) external view returns (uint256)',
+  'function tokenURI(uint256 tokenId) public view returns (string uri)'
+];
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+) {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+
+  // 1st, get the balance of the token
+  const callWalletToBalanceOf = new Multicaller(network, provider, abi, {
+    blockTag
+  });
+  for (const walletAddress of addresses) {
+    callWalletToBalanceOf.call(walletAddress, options.address, 'balanceOf', [
+      walletAddress
+    ]);
+  }
+  const walletToBalanceOf: Record<
+    string,
+    BigNumber
+  > = await callWalletToBalanceOf.execute();
+
+  // 2nd, get the first tokenId for each address
+  const callWalletToFirstTokenID = new Multicaller(network, provider, abi, {
+    blockTag
+  });
+  for (const [walletAddress, count] of Object.entries(walletToBalanceOf)) {
+    if (count.toNumber() > 0) {
+      callWalletToFirstTokenID.call(
+        walletAddress.toString(),
+        options.address,
+        'tokenOfOwnerByIndex',
+        [walletAddress, 0]
```

Comment:
Here you are trying to get 0 index token? it should be based on the balance of the wallet right?

Reply:
I mean why only first token ID?

Reply:
in any strategy, there should be maximum 5 requests, I can't get all nfts tokenURI.
so I decided to use the only 1st nft's metadata.
and `tokenOfOwnerByIndex(address, 0)` gives an error if balance is 0.
so I checked balance, and called `tokenOfOwnerByIndex` if there is nft.

Reply:
I guess, there should be 3 requests for getting `balanceOf`, `tokenOfOwnerByIndex` and `tokenURI`.
after that, I can fetch uri data maximum 2 times.
so I decided to use the 1st nft only.

Reply:
Do you have any suggestion?

Reply:
> in any strategy, there should be maximum 5 requests, I can't get all nfts tokenURI. so I decided to use the only 1st nft's metadata. and `tokenOfOwnerByIndex(address, 0)` gives an error if balance is 0. so I checked balance, and called `tokenOfOwnerByIndex` if there is nft.